app/screens/home/channel_list/categories_list/header/index.ts

```diff
@@ -25,6 +25,7 @@ const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
 
     const canJoinChannels = combineLatest([currentUser, team]).pipe(
         switchMap(([u, t]) => observePermissionForTeam(database, t, u, Permissions.JOIN_PUBLIC_CHANNELS, true)),
+        distinctUntilChanged(),
```

Comment:
These changes help with the categories header not being re-rendered more than needed.

observePermissionForTeam already check for distincUntilChanged, but the team changing recreates the observer, so we need an extra distinctUntilChanged here.